File: 787-cheapest-flights-within-k-stops/787-cheapest-flights-within-k-stops.py
class Solution:
    def findCheapestPrice(self, n: int, flights: List[List[int]], src: int, dst: int, k: int) -> int:
        
        # Bellman-Ford #
        
        inf = (int)(1e9+7)
        dp = [ [inf for _ in range(n)] for __ in range(k+2)]
        
        for i in range(0,k+2):
            dp[i][src]=0
        
        for i in range(1,k+2):
            for [u, v, w] in flights:
                dp[i][v]=min(dp[i][v], dp[i-1][u] + w)
        
        # return dp
        return (-1 if dp[k+1][dst]==inf else dp[k+1][dst]) 
    

File: 787-cheapest-flights-within-k-stops/787-cheapest-flights-within-k-stops.cpp
class Solution {
public:
    int findCheapestPrice(int n, vector<vector<int>>& flights, int src, int dst, int k) {
        
        const int inf=1e9;
        int dp[k+2][n];
        
        for(int i=0;i<k+2;++i)
            for(int v=0;v<n;++v)
                dp[i][v]=inf;
        for(int i=0;i<k+2;++i) dp[i][src]=0;
        
        for(int i=1;i<k+2;++i){
            for(auto &edge: flights){
                int u=edge[0],v=edge[1],w=edge[2];
                
                dp[i][v]=min({dp[i][v], dp[i-1][u]+w});
            }
        }
        
        return ((dp[k+1][dst]==inf)?-1:dp[k+1][dst]);
    }
};